lib/services/access_list.go

```diff
@@ -276,9 +418,21 @@ func (a AccessListMembershipChecker) IsAccessListMember(ctx context.Context, ide
 	}
 
 	member, err := a.members.GetAccessListMember(ctx, accessList.GetName(), username)
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
 	if trace.IsNotFound(err) {
-		// The member has not been found, so we know they're not a member of this list.
-		return trace.NotFound("user %s is not a member of the access list", username)
+		// try find if the user could be a member of any lists by recursing
+		err := a.recursiveIsAccessListMemberCheck(ctx, identity, accessList)
+		if trace.IsNotFound(err) {
+			// The member has not been found, so we know they're not a member of this list.
+			return trace.NotFound("user %s is not a member of the access list", username)
+		}
+		if err != nil {
+			// Some other error has occurred
+			return trace.Wrap(err)
+		}
+		return nil
```

Comment:
The `UserMeetsRequirements` check in https://github.com/gravitational/teleport/pull/38738/files#diff-57e71bdbf9e192452bb9a2a45cb751930330be9ffe63caf97264c1c5aa324087R446 

is omitted so if the parent access list has some constraint about membership this check will be omitted. 
Resulting that access list constraint can be bypassed by nested access list behavior.